preview-src/timeline.tsx

```diff
@@ -0,0 +1,117 @@
+import * as React from 'react';
+
+import { Comment } from '../src/common/comment';
+import { TimelineEvent, isReviewEvent, isCommitEvent, isCommentEvent, isMergedEvent, isAssignEvent, ReviewEvent, CommitEvent, CommentEvent, MergedEvent, AssignEvent } from '../src/common/timelineEvent';
+import { commitIcon, mergeIcon } from './icon';
+import { Avatar, AuthorLink } from './user';
+import { groupBy } from '../src/common/utils';
+import { Spaced } from './space';
+import Timestamp from './timestamp';
+import { CommentView } from './comment';
+import Diff from './diff';
+
+export const Timeline = ({ events }: { events: TimelineEvent[] }) =>
+	<>{
+		events.map(event =>
+			// TODO: Maybe make TimelineEvent a tagged union type?
+			isCommitEvent(event)
+				? <CommitEventView key={event.id} {...event} />
```

Comment:
I'm wondering if each of these views should be in its own file, perhaps grouped together in a folder?

Reply:
I mean... maybe. It's kindof on the edge for me, because they're pretty small and they're only referenced by the timeline.

We could pull out the thread view? That seems like it could be useful elsewhere at some point, and it's most of the complexity.